game/state/savemanager.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+#include "framework/serialization/serialize.h"
+#include "game/state/gamestate.h"
+#include <future>
+#include <tuple>
+
+class SerializationNode;
+class GameState;
+namespace OpenApoc
+{
+
+enum SaveType
+{
+	Manual = 0,
+	Auto = 1,
+	Quick = 2
+};
+
+/* Metadata of save archive displayed in savemenu	*/
+class SaveMetadata
+{
+  private:
+	UString name;
+	UString file;
+	UString difficulty;
+	time_t creationDate;
+	SaveType type;
+	unsigned long gameTicks;
+
+  public:
+	SaveMetadata();
+	~SaveMetadata();
+	SaveMetadata(UString name, UString file, time_t creationDate, SaveType type,
+	             const sp<GameState> gameState);
+	SaveMetadata(const SaveMetadata &metdata, time_t creationDate, const sp<GameState> gameState);
+
+	/* Deserialize given manifest document	*/
+	bool deserializeManifest(const sp<SerializationArchive> archive, const UString &saveFileName);
+
+	bool serializeManifest(const sp<SerializationArchive> archive) const;
+
+	/* Creation date of save file (unix epoch)	*/
+	const time_t getCreationDate() const;
```

Comment:
'const' return types doesn't mean anything for non-pointer/reference - maybe remove that?